### PR TITLE
Add BoundaryEvent cancellation hook for telemetry

### PR DESCRIPTION
**Pull request by ygree**
This new internal hook for telemetry. So, it knows when an event won't be executed to properly release a retained span continuation.

### CODE REVIEW

File: akka-stream/src/main/mima-filters/2.6.15.backwards.excludes/30246-boundary-event-cancellation-hook.excludes [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Changes to internal/private
+ProblemFilters.exclude[ReversedMissingMethodProblem]("akka.stream.impl.fusing.ActorGraphInterpreter#BoundaryEvent.cancel")
```
Comment (johanandren) [inline akka-stream/src/main/mima-filters/2.6.15.backwards.excludes/30246-boundary-event-cancellation-hook.excludes:2]: Should be in 2.6.14 backwards excludes, latest released version. Curious that the CI isn't failing.

**Review by johanandren — COMMENTED** [review]
Change looks good, mima filter needs directory rename

File: akka-stream/src/main/mima-filters/2.6.15.backwards.excludes/30246-boundary-event-cancellation-hook.excludes (hunk 1 above)
Comment (octonato) [reply akka-stream/src/main/mima-filters/2.6.15.backwards.excludes/30246-boundary-event-cancellation-hook.excludes:2]: To be honest, I never know which versions to pick. I always have to look around. 

I think the reason is that `2.6.15.backwards.excludes` reads as: what are the exclusions that we version 2.6.15 had to to add. 

No idea why CI didn't complain.

**Review by octonato — APPROVED** [review]
LGTM, after ...

```shell
mv 2.6.15.backwards.excludes/* 2.6.14.backwards.excludes/
rm -r 2.6.15.backwards.excludes
```

**Review by johanandren — APPROVED** [review]

**Pull request merged**